services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/monitor/core.py

```diff
@@ -0,0 +1,412 @@
+# here services can be registered to be monitored for their
+# current status. Events are triggered base on what happens
+# Monitoring for:
+# - servie available
+#
+#
+# Handlers to subscribe to this thing
+
+import asyncio
+import logging
+import traceback
+from asyncio import Lock, sleep
+from copy import deepcopy
+from typing import Deque, Dict, Optional
+
+from async_timeout import timeout
+from fastapi import FastAPI, Request
+from models_library.projects_nodes_io import NodeID
+
+from ....core.settings import (
+    DynamicServicesMonitoringSettings,
+    DynamicServicesSettings,
+    DynamicSidecarSettings,
+)
+from ....models.schemas.dynamic_services import RunningServiceDetails
+from ..docker_utils import (
+    ServiceLabelsStoredData,
+    are_all_services_present,
+    get_dynamic_sidecar_state,
+    get_dynamic_sidecars_to_monitor,
+    remove_dynamic_sidecar_network,
+    remove_dynamic_sidecar_stack,
+)
+from ..exceptions import DynamicSidecarError, DynamicSidecarNotFoundError
+from ..parse_docker_status import ServiceState, extract_containers_minimim_statuses
+from .dynamic_sidecar_api import (
+    DynamicSidecarClient,
+    get_api_client,
+    update_dynamic_sidecar_health,
+)
+from .handlers import REGISTERED_EVENTS
+from .models import DynamicSidecarStatus, LockWithMonitorData, MonitorData
+from .utils import AsyncResourceLock
+
+logger = logging.getLogger(__name__)
+
+
+async def apply_monitoring(app: FastAPI, monitor_data: MonitorData) -> None:
+    """
+    fetches status for service and then processes all the registered events
+    and updates the status back
+    """
+    dynamic_services_settings: DynamicServicesSettings = (
+        app.state.settings.dynamic_services
+    )
+    initial_overall_status = deepcopy(monitor_data.dynamic_sidecar.overall_status)
+
+    # checks if service is still present, if not removes
+    # self from monitor and log it as warning
+    # use a property to check that these are not here
+    if (  # do not refactor, second part of and condition is skiped most times
```

Comment:
you might need to refactor that comment. I could not understand its meaning. ;)

Reply:
you are correct, done